Fix: Improve selection contrast in mapper room list and area dropdown

mpconley opened the pull request:
#### Brief overview of PR changes/additions

Fixes poor contrast in mapper selection lists by explicitly setting the application palette on the room selection widget and area dropdown combobox. This ensures proper highlight colors (blue background with white text) are used in both light and dark themes.

#### Motivation for adding to Mudlet

Addresses issue #8219 where selected items in the mapper's room selection list and area dropdown had very poor contrast:
- Dark mode: black text on dark grey background
- Light mode: light grey text on white background

This made it extremely difficult to identify which rooms or areas were selected.

#### Other info (issues closed, discussion etc)

Closes #8219

The fix applies the application-level palette to both widgets, ensuring they inherit the proper `QPalette::Highlight` and `QPalette::HighlightedText` colors configured by Mudlet's `DarkTheme` class and system defaults.

mpconley committed:
initial

Kebap commented:
> Hey there! Thanks for helping Mudlet improve. 🌟
>     * windows 64 bit: (download pending, check back soon!)

Here it is: https://github.com/Mudlet/Mudlet/actions/runs/19449063298/artifacts/4595653780

Thanks for giving this a shot. However the results are not really helpful, yet.

Test result with system default (dark):

<img width="383" height="469" alt="grafik" src="https://github.com/user-attachments/assets/d90821e3-57cf-49b9-8700-34a24b856dbf" />


Switched Mudlet to dark mode:

<img width="368" height="578" alt="grafik" src="https://github.com/user-attachments/assets/6e35cbe8-ae53-4398-a929-1f83ebd98196" />

See differences in area selection dropdown, but not in room selection list..?!

However both contrast options still seem unwieldy to me now. 

Compare positive example in the original issue.

Switched Mudlet to light mode:

<img width="374" height="578" alt="grafik" src="https://github.com/user-attachments/assets/2b778f92-6cac-4cf9-95fa-7f70d98b7fe2" />


Note how the selection was not helped much, but instead the map controls are now invisible..!

mpconley committed:
Merge branch 'development' into fix-mapper-selection-list-contrast

mpconley committed:
Fix: Improve selection contrast in mapper selection UI\n\n- Apply app palette to area dropdown via combo palette\n- Move room selection list palette to T2DMap::init() so it overrides parent palette\n- Remove dlgMapper-wide palette override to avoid dimming map controls\n\nAddresses #8219 and responds to PR feedback with targeted palette use.

mpconley commented:
> However the results are not really helpful, yet.

Thanks. Let's try again after this build, please. Mac does not look the same, but parsing detail from your pictures helps.

mpconley committed:
Merge branch 'development' into fix-mapper-selection-list-contrast

mpconley committed:
Address CodeRabbit review: improve palette handling

- Apply palette to comboBox_showArea->view() for better Windows compatibility
- Update comment in T2DMap.cpp for clarity about application palette usage
- Add palette refresh for live theme switching in HostManager::changeAllHostColour()
  - Updates comboBox_showArea and its view when themes change
  - Updates mMultiSelectionListWidget when themes change

These changes ensure selection contrast remains correct when users switch
themes at runtime, not just at initialization.

vadi2 commented:
@kebap any difference with the new build?

Kebap commented:
Testing (commit https://github.com/Mudlet/Mudlet/commit/a4443cee4590d4f95af4b0a5f7976f19000318d5) on Win 11

System default (dark):

<img width="354" height="586" alt="grafik" src="https://github.com/user-attachments/assets/30699ab2-d7de-4dab-8efd-5e0eb9d017ec" />



Switching Mudlet to light mode:

<img width="356" height="581" alt="grafik" src="https://github.com/user-attachments/assets/71d8cad7-7a28-4941-8f18-0ee2cacd5ed0" />



Switching Mudlet to dark mode:

<img width="362" height="581" alt="grafik" src="https://github.com/user-attachments/assets/994ad478-da3c-4af7-a565-e52f8d2e80e0" />



Verdict: The situation did not improve (much / at all?)

mpconley committed:
Merge branch 'development' into fix-mapper-selection-list-contrast

vadi2 commented:
I notice you're using a stylesheet to customise Mudlets UI. Does this happen on a new profile without the customisation loaded @Kebap?

mpconley committed:
Fix: Use QSS-based selection styling for cross-platform reliability

- Replace palette-based approach with Qt Style Sheets
- Apply stylesheet directly to dropdown view instead of combobox to avoid affecting sibling widgets
- Add separate styling for active and inactive selection states
- Use palette(highlight) and palette(highlighted-text) to maintain theme compatibility
- Prevents map controls from becoming invisible (issue from previous approach)
- Should resolve selection contrast issues on Windows while maintaining macOS functionality

mpconley committed:
Style: Fix cpplint whitespace/parens warnings

Move closing parentheses to previous line as per cpplint standards

mpconley committed:
Refactor: Use single qsl() with implicit string concatenation for stylesheets

Following Mudlet's standard pattern for multiline strings, refactor the
QSS stylesheets to use a single qsl() call with implicit C++ string
literal concatenation instead of multiple qsl() calls with + operators.

This matches the pattern used throughout the codebase (e.g., in
dlgRoomProperties.cpp) and should resolve cpplint warnings while
maintaining readability.

vadi2 commented on src/dlgMapper.cpp:
```diff
@@ -89,6 +89,20 @@ dlgMapper::dlgMapper( QWidget * parent, Host * pH, TMap * pM )
+    // Using QSS for reliable cross-platform behavior, especially on Windows
+    // Apply only to the dropdown view to avoid affecting the combobox button itself
+    if (auto* view = comboBox_showArea->view()) {
+        view->setStyleSheet(
```
Unfortunately this will conflict with https://wiki.mudlet.org/w/Manual:Lua_Functions#setProfileStyleSheet

I'd like to see if this issue is due to a custom stylesheet to begin with

mpconley replied:
Backed out that change

mpconley committed:
Revert: Remove QSS-based selection styling approach

Per reviewer feedback, reverting to the palette-based approach.
This removes the Qt Style Sheet implementation and restores the
setPalette() calls on both the room selection list widget and
area dropdown combobox.

Reverts commits:
- f99829e9 (Refactor: Use single qsl() with implicit string concatenation)
- 16ad6a65 (Style: Fix cpplint whitespace/parens warnings)
- 3d8f4118 (Fix: Use QSS-based selection styling)

Back to state at commit 238cc7b5.

vadi2 commented:
Room selection and area dropdown looks OK on dark mode in windows 11 without any custom themes applied:

<img width="640" height="400" alt="image" src="https://github.com/user-attachments/assets/6c2de6e9-5148-4a94-83a8-fc29a5b4b75e" />

mpconley committed:
Merge branch 'development' into fix-mapper-selection-list-contrast

Kebap commented:
I will double check that. How many of your 3 rooms are selected? Does not seem very obvious or OK to me..

vadi2 commented:
All are selected. This is with 2 and 4 selected:

<img width="640" height="400" alt="image" src="https://github.com/user-attachments/assets/5b8c32a0-d5e8-4a83-ada4-83548cacfa55" />

vadi2 commented:
I think we should close this one, see https://github.com/Mudlet/Mudlet/issues/8219#issuecomment-3576663780
